terraform/modules/app/main.tf

```diff
@@ -0,0 +1,87 @@
+# modules/app defines a set of resources which are essential to every
+# microservice in the system. If a resource is not common to all apps,
+# it probably doesn't belong here.
+
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 2.69"
+    }
+  }
+}
+
+locals {
+  container_definitions = [
+    {
+      "name" : "envoy",
+      # TODO: don't hardcode the version; track stable Envoy
+      # TODO: control when Envoy updates happen (but still needs to be automatic)
+      # TODO: don't hardcode the region
+      "image" : "840364872350.dkr.ecr.eu-west-1.amazonaws.com/aws-appmesh-envoy:v1.15.0.0-prod",
+      "user" : "1337",
+      "environment" : [
+        { "name" : "APPMESH_VIRTUAL_NODE_NAME", "value" : "mesh/${var.mesh_name}/virtualNode/${var.service_name}" }
+      ],
+      "essential" : true,
+      "logConfiguration" : {
+        "logDriver" : "awslogs",
+        "options" : {
+          "awslogs-create-group" : "true",
+          "awslogs-group" : "awslogs-fargate",
+          "awslogs-region" : "eu-west-1", # TODO: hardcoded
+          "awslogs-stream-prefix" : "awslogs-${var.service_name}-envoy"
+        }
+      }
+    }
+  ]
+}
+
+resource "aws_ecs_task_definition" "service" {
+  family                   = var.service_name
+  requires_compatibilities = ["FARGATE"]
+  container_definitions    = jsonencode(concat(var.container_definitions, local.container_definitions))
+  network_mode             = "awsvpc"
+  cpu                      = var.cpu
+  memory                   = var.memory
+  task_role_arn            = var.task_role_arn
+  execution_role_arn       = var.execution_role_arn
+
+  proxy_configuration {
+    type           = "APPMESH"
+    container_name = "envoy"
+
+    properties = {
+      AppPorts         = var.container_ingress_port
+      EgressIgnoredIPs = "169.254.170.2,169.254.169.254" # TODO: no longer required (try omitting, might need to stay but empty?)
+      # TODO: what are these magic numbers and are they necessary?
```

Comment:
These magic numbers are sadly necessary. Envoy listens on 15000 and 15001. The IgnoredUID is required but set arbitrarily as "1337" since this is also used in the AWS documentation.

https://docs.aws.amazon.com/AmazonECS/latest/userguide/example_task_definitions.html#example_task_definition-containerdependency

https://github.com/awsdocs/amazon-ecs-developer-guide/blob/master/doc_source/appmesh-getting-started.md#-json-

Reply:
Thanks, that confirms my suspicions! :D I'll follow up with a PR to clean up some of those TODOs.